mongo/mongo_test.go

```diff
@@ -72,52 +72,45 @@ func TestMongoHelpers(t *testing.T) {
 	t.Run("transform and ensure ID", func(t *testing.T) {
 		t.Run("newly added _id should be first element", func(t *testing.T) {
 			doc := bson.D{{"foo", "bar"}, {"baz", "qux"}, {"hello", "world"}}
-			want := bsonx.Doc{
-				{"_id", bsonx.Null()}, {"foo", bsonx.String("bar")},
-				{"baz", bsonx.String("qux")}, {"hello", bsonx.String("world")},
-			}
 			got, id, err := transformAndEnsureID(bson.DefaultRegistry, doc)
 			assert.Nil(t, err, "transformAndEnsureID error: %v", err)
 			oid, ok := id.(primitive.ObjectID)
 			assert.True(t, ok, "expected returned id type %T, got %T", primitive.ObjectID{}, id)
-			want[0] = bsonx.Elem{"_id", bsonx.ObjectID(oid)}
-			assert.Equal(t, got, want, "expected document %v, got %v", got, want)
-		})
-		t.Run("existing _id should be first element", func(t *testing.T) {
```

Comment:
I removed this test. The new `transformAndEnsureID()` does not seem to move the `_id` field to be the first element in the document. Not sure if this is a bug or is intended behavior.

Reply:
I can't find anywhere in the crud spec where it says to move it, so i think this is fine.

Reply:
I looked up the history of that test. It was added with https://github.com/mongodb/mongo-go-driver/commit/5514df920614a2ffba26ec3539d8509e8ad2344e and references ticket https://jira.mongodb.org/browse/GODRIVER-660. From the ticket:
> Server engineering team recommends that for performance reasons the _id field should be the first one in any document being inserted.  Currently auto-generated _id fields are appended rather than prepended, making them the last field instead of first.

Is that `_id`-first performance improvement still accurate? If so, I think we should keep this test and requirement (and maybe add it to the spec somewhere as a recommendation). If the order of `_id` no longer has a server-side performance impact, then the test is safe to remove.

Reply:
Thanks for looking into the original ticket! Discussed offline: it was indeed intentional to stop moving the `_id` to be the first field in the document. We'd like to leave this behavior as is to avoid re-ordering the user's document. 

It is still true that an `_id` field that comes anywhere but first can create a slight loss in performance, so I made [GODRIVER-1987](https://jira.mongodb.org/browse/GODRIVER-1978) to document this for users.

Reply:
Sounds good 👍